charts/drupal/templates/_helpers.tpl

```diff
@@ -235,40 +235,67 @@ done
 -f /app/web/sites/default/files/_installing
 {{- end -}}
 
+
 {{- define "drupal.post-release-command" -}}
-set -e
+  set -e
 
-{{ if and .Release.IsInstall .Values.referenceData.enabled -}}
-{{ include "drupal.import-reference-files" . }}
-{{- end }}
+  {{ if and .Release.IsInstall .Values.referenceData.enabled -}}
+    {{ include "drupal.import-reference-files" . }}
+  {{- end }}
 
-{{ include "drupal.wait-for-db-command" . }}
+  {{ include "drupal.wait-for-db-command" . }}
 
-{{ if .Release.IsInstall }}
-touch /app/web/sites/default/files/_installing
-{{- if .Values.referenceData.enabled }}
-{{ include "drupal.import-reference-db" . }}
-{{- end }}
-{{- end }}
+  {{ if .Release.IsInstall }}
+    touch /app/web/sites/default/files/_installing
+    {{- if .Values.referenceData.enabled }}
+      {{ include "drupal.import-reference-db" . }}
+    {{- end }}
+  {{- end }}
 
-{{ if .Values.elasticsearch.enabled }}
-{{ include "drupal.wait-for-elasticsearch-command" . }}
-{{ end }}
+  {{ if .Values.elasticsearch.enabled }}
+    {{ include "drupal.wait-for-elasticsearch-command" . }}
+  {{ end }}
 
-{{ if .Release.IsInstall }}
-{{ .Values.php.postinstall.command}}
-rm /app/web/sites/default/files/_installing
-{{ end }}
-{{ .Values.php.postupgrade.command}}
+  {{ if .Release.IsInstall }}
+    {{ .Values.php.postinstall.command }}
+    rm /app/web/sites/default/files/_installing
+  {{ end }}
+  {{ .Values.php.postupgrade.command}}
 
-# Wait for background imports to complete.
-wait
+  # Wait for background imports to complete.
+  wait
 
-{{- if and .Values.referenceData.enabled .Values.referenceData.updateAfterDeployment }}
-{{- if eq .Values.referenceData.referenceEnvironment .Values.environmentName }}
-{{ include "drupal.extract-reference-data" . }}
-{{- end }}
+  {{- if and .Values.referenceData.enabled .Values.referenceData.updateAfterDeployment }}
+    {{- if eq .Values.referenceData.referenceEnvironment .Values.environmentName }}
+      {{ include "drupal.extract-reference-data" . }}
+    {{- end }}
+  {{- end }}
 {{- end }}
+
+{{- define "drupal.restore-backup-command" -}}
+  set -e
+
+  {{ include "drupal.wait-for-db-command" . }}
+
+  # Make backup of current deployment
+  {{ include "drupal.backup-command" . }}
+  
+  # Restore files from targeted backup
+  {{ include "drupal.import-backup-files" . }}
+
+  touch /app/web/sites/default/files/_installing
+
+  # Restore db from targeted backup
+  {{ include "drupal.import-backup-db" . }}
+
+  {{ if .Values.elasticsearch.enabled }}
+    {{ include "drupal.wait-for-elasticsearch-command" . }}
+  {{ end }}
+
+  # Running custom commands after restored backup
+  {{ .Values.php.postRestoreCommand }}
```

Comment:
At this point we still need to run `{{ .Values.php.postupgrade.command}}`, for example to run database updates.

Reply:
Added right before executing postRestoreCommand